File: OnlyObjcAtStart/OnlyObjcAtStart/OnlyObjcAtStart/RootViewController.m
//
//  RootViewController.m
//  LazySwift
//
//  Created by Nikolay Lihogrud
//  Copyright © 2017 Yandex LLC. All rights reserved.
//

#import "RootViewController.h"

@interface RootViewController ()

@property(nonatomic, weak) id<SearchFacade> searchFacade;
@property(nonatomic, weak) id<RoutingFacade> routingFacade;

@property(nonatomic, strong) UIButton *searchButton;
@property(nonatomic, strong) UIButton *routeButton;

@end

@implementation RootViewController

- (instancetype)initWithSearchFacade:(id<SearchFacade>)searchFacade routingFacade:(id<RoutingFacade>)routingFacade {
    self = [super initWithNibName:nil bundle:nil];

    self.searchFacade = searchFacade;
    self.routingFacade = routingFacade;

    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];

    self.view.backgroundColor = [UIColor whiteColor];

    self.searchButton = [UIButton buttonWithType:UIButtonTypeSystem];
    [self.view addSubview:self.searchButton];
    self.searchButton.frame = CGRectMake(100, 100, 100, 100);
    [self.searchButton setTitle:@"Search" forState:UIControlStateNormal];
    [self.searchButton addTarget:self action:@selector(onSearchButtonTapped) forControlEvents:UIControlEventTouchUpInside];


    self.routeButton = [UIButton buttonWithType:UIButtonTypeSystem];
    [self.view addSubview:self.routeButton];
    self.routeButton.frame = CGRectMake(100, 300, 100, 100);
    [self.routeButton setTitle:@"Route" forState:UIControlStateNormal];
    [self.routeButton addTarget:self action:@selector(onRoutingButtonTapped) forControlEvents:UIControlEventTouchUpInside];
}

- (void)onSearchButtonTapped {
    [self.searchFacade performSearch];
}

- (void)onRoutingButtonTapped {
    [self.routingFacade performRouting];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: OnlyObjcAtStart/Original/Original/RootViewController.swift
//
//  RootViewController.swift
//  Original
//
//  Created by Nikolay Lihogrud
//  Copyright © 2017 Yandex LLC. All rights reserved.
//

import UIKit

class RootViewController: UIViewController {

    fileprivate(set) weak var searchFacade: SearchFacade?
    fileprivate(set) weak var routingFacade: RoutingFacade?

    fileprivate let searchButton = UIButton(type: .system)
    fileprivate let routingButton = UIButton(type: .system)

    init(searchFacade: SearchFacade, routingFacade: RoutingFacade) {
        self.searchFacade = searchFacade
        self.routingFacade = routingFacade

        super.init(nibName: nil, bundle: nil)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func viewDidLoad() {
        super.viewDidLoad()

        view.backgroundColor = .white

        view.addSubview(searchButton)
        searchButton.frame = CGRect(x: 100, y: 100, width: 100, height: 100)
        searchButton.setTitle("Search", for: .normal)
        searchButton.addTarget(self, action: #selector(onSearchButtonTapped), for: .touchUpInside)

        view.addSubview(routingButton)
        routingButton.frame = CGRect(x: 100, y: 300, width: 100, height: 100)
        routingButton.setTitle("Route", for: .normal)
        routingButton.addTarget(self, action: #selector(onRoutingButtonTapped), for: .touchUpInside)

    }

    @objc func onSearchButtonTapped() {
        searchFacade?.performSearch()
    }

    @objc func onRoutingButtonTapped() {
        routingFacade?.performRouting()
    }

}

